eng/scripts/Language-Settings.ps1

```diff
@@ -23,8 +23,29 @@ function Get-javascript-EmitterName() {
   return "@azure-tools/typespec-ts"
 }
 
+function Get-Relative-Generated-Folder([string]$projectDirectory) {
+  $tspLocationFile = Resolve-Path "$projectDirectory/tsp-location.yaml"
+  Write-Host "Reading configuration from $tspLocationFile"
+  $tspLocationConfig = Get-Content -Path $tspLocationFile -Raw | ConvertFrom-Yaml
+  $relativeTempDirectory = $tspLocationConfig["directory"]
+  $start = $relativeTempDirectory.LastIndexOf("/")
+  $relativeTempDirectory = $relativeTempDirectory.Substring($start + 1)
+  $tspConfigFile = Resolve-Path "$projectDirectory/TempTypeSpecFiles/$relativeTempDirectory/tspconfig.yaml"
+  Write-Host "Reading tsp configuration from $tspConfigFile"
+  $tspConfigConfig = Get-Content -Path $tspConfigFile -Raw | ConvertFrom-Yaml
```

Comment:
We would like to consolidate all the interactions with `tspconfig.yaml` in the common scripts. +@weshaggard to comment.

Can you share the reason of adding `relativeGeneratedFolder` config? Does it need to be customized by service team? Dotnet actually always drop the generated code under `{package-dir}/src`, can JS use fixed relative path?

Reply:
`relativeGeneratedFolder` is needed to configure where to write the generated code files. This is needed in JS because we don't want to write the generated files in `{package-dir}/src` but rather in `{package-dir}/sources/generated`. We then use a special tool to augment the generated source files with custom code and write the result in `{package-dir}/src`.

Reply:
If is a common convention then why not just do it based off the base directory? As @raych1 mentioned I want to avoid anyone else directly parsing the tspconfig file.

Reply:
Agreed with the direction of this thread that we should be smarter about how the emitter places files on disk. I have reverted the changes to this file so we can get the typespec updates into main